denso_robot.py

```diff
@@ -100,3 +110,8 @@ def check_sample_mounted(self, mount, puck_pos, pin_pos):  # is the correct samp
                     return UNMOUNT_FAILURE
         else:
             return MOUNT_STEP_SUCCESSFUL  # always successful if robot is not online
+
+# This is a generic interface to give robots access to the robot_lib api
+class OphydRobot(DensoRobot):
+    def __init__(self):
+        self.robot = robot
```

Comment:
robot is not defined here - did you mean to pass in a robot?

but if you do that, then to access any functions, you would need to do something like `robot.robot.dismount()`.

did you actually mean to use a mixin or something similar so that you can add the canExchangeSamples in your class?